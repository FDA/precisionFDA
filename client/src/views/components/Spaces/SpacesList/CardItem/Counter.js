import React from 'react'
import PropTypes from 'prop-types'


import { getSpacesIcon } from '../../../../../helpers/spaces'
import Icon from '../../../Icon'


const Counter = ({ type, counter }) => {
  const icon = getSpacesIcon(type)

  return (
    <div className="spaces-list-card-data__counter">
      <div>
        <Icon icon={icon} />
      </div>
      <div>{counter}</div>
    </div>
  )
}

Counter.propTypes = {
  type: PropTypes.string.isRequired,
  counter: PropTypes.number.isRequired,
}

export default Counter
