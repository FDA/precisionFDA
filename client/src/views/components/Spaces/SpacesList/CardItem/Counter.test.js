import React from 'react'
import { shallow } from 'enzyme'

import Counter from './Counter'


describe('<Counter />', () => {
  it('matches snapshot', () => {
    const wrapper = shallow(<Counter type={'member'} counter={100} />)

    expect(wrapper).toMatchSnapshot()
  })
})
