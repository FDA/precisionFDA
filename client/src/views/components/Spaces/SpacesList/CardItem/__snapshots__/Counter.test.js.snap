// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counter /> matches snapshot 1`] = `
<div
  className="spaces-list-card-data__counter"
>
  <div>
    <Icon
      icon=""
    />
  </div>
  <div>
    100
  </div>
</div>
`;
